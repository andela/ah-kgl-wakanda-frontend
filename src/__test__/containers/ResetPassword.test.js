import React from 'react';
import { shallow, mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ResetPasswordRedux, { ResetPassword } from '../../Containers/ResetPassword/ResetPassword';

const mockStore = configureMockStore([thunk]);

const props = {
  onResetPassword: jest.fn(),
  emailSent: false,
  message: '',
};

describe('render ResetPassword component', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      resetPassword: { emailSent: false, error: false, message: '', onResetPassword: jest.fn() },
    });
  });

  it('should have Input component', async () => {
    const wrapper = shallow(<ResetPasswordRedux store={store} {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not have Input component', async () => {
    const props1 = { ...props, emailSent: true };
    const wrapper = shallow(<ResetPassword store={store} {...props1} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Input').length).toBe(0);
  });

  it('should update the email value in the state ', async () => {
    const wrapper = mount(<ResetPassword store={store} {...props} />);
    wrapper
      .find('input')
      .simulate('input', { target: { name: 'email', value: 'email@gmail.com' } });
    expect(wrapper.instance().state.email).toBe('email@gmail.com');
  });

  it('should called a onResetPassword', async () => {
    const wrapper = mount(<ResetPassword store={store} {...props} />);
    wrapper
      .find('input')
      .simulate('input', { target: { name: 'email', value: 'email@gmail.com' } });
    wrapper.find('form').simulate('submit');
    expect(props.onResetPassword).toHaveBeenCalledWith('email@gmail.com');
  });
});
