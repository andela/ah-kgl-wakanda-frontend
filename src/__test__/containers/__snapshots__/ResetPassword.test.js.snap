// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ResetPassword component should have Input component 1`] = `ShallowWrapper {}`;

exports[`render ResetPassword component should not have Input component 1`] = `ShallowWrapper {}`;
